File: src/Controls/KarveControls/DataGrid/DataGridHelpers/DataGridDecimalColumn.cs
using Telerik.WinControls.UI;

namespace KarveControls.DataGrid.DataGridHelpers
{
    public class DataGridDecimalColumn : GridViewDecimalColumn
    {

        public DataGridDecimalColumn()
        {
            this.NumeroDecimales = 2;
        }

        #region "Variables"

        int _Item;
        string _Campo;
        string _Tabla;
        string _ExpresionBd;
        #endregion
        System.Drawing.Color _BackColor = System.Drawing.Color.White;

        #region "Propiedades"

        public int Item
        {
            get { return _Item; }
            set { _Item = value; }
        }

        public string Tabla
        {
            get { return _Tabla; }
            set { _Tabla = value; }
        }

        public string Campo
        {
            get { return _Campo; }
            set { _Campo = value; }
        }

        public string AliasCampo
        {
            get { return FieldName; }
            set { FieldName = value; }
        }

        public string ExpresionBd
        {
            get { return _ExpresionBd; }
            set { _ExpresionBd = value; }
        }

        public System.Drawing.Color BackColor
        {
            get { return _BackColor; }
            set { _BackColor = value; }
        }

        public override bool ReadOnly
        {
            get { return base.ReadOnly; }
            set
            {
                base.ReadOnly = value;
                if (value)
                {
//                _BackColor = ColorSel;
                }
            }
        }

        public int NumeroDecimales
        {
            get { return this.DecimalPlaces; }
            set
            {
                this.DecimalPlaces = value;
                this.FormatString = "{0:N" + value + "}";
            }
        }
        #endregion

    }
}

File: documentation/Karve/CustomControls/DataGrid/Clases/DataGridColumn/Types/DataGridDecimalColumn.vb
Imports Telerik.WinControls.UI
Imports VariablesGlobales
Imports Karve.ConfiguracionApp

Public Class DataGridDecimalColumn
    Inherits GridViewDecimalColumn

    Public Sub New()
        Me.NumeroDecimales = 2
    End Sub

#Region "Variables"

    Dim _Item As Integer
    Dim _Campo As String
    Dim _Tabla As String
    Dim _ExpresionBd As String
    Dim _BackColor As System.Drawing.Color = Drawing.Color.White
#End Region

#Region "Propiedades"

    Property Item As Integer
        Get
            Return _Item
        End Get
        Set(value As Integer)
            _Item = value
        End Set
    End Property

    Property Tabla As String
        Get
            Return _Tabla
        End Get
        Set(value As String)
            _Tabla = value
        End Set
    End Property

    Property Campo As String
        Get
            Return _Campo
        End Get
        Set(value As String)
            _Campo = value
        End Set
    End Property

    Property AliasCampo As String
        Get
            Return FieldName
        End Get
        Set(value As String)
            FieldName = value
        End Set
    End Property

    Property ExpresionBd As String
        Get
            Return _ExpresionBd
        End Get
        Set(value As String)
            _ExpresionBd = value
        End Set
    End Property

    Property BackColor As System.Drawing.Color
        Get
            Return _BackColor
        End Get
        Set(value As System.Drawing.Color)
            _BackColor = value
        End Set
    End Property

    Public Overrides Property [ReadOnly] As Boolean
        Get
            Return MyBase.[ReadOnly]
        End Get
        Set(value As Boolean)
            MyBase.[ReadOnly] = value
            If value Then
                _BackColor = ColorSel
            End If
        End Set
    End Property

    Public Property NumeroDecimales As Integer
        Get
            Return Me.DecimalPlaces
        End Get
        Set(value As Integer)
            Me.DecimalPlaces = value
            Me.FormatString = "{0:N" & value & "}"
        End Set
    End Property
#End Region

End Class
